dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java

```diff
@@ -144,6 +151,39 @@ public final Response getCategories(@Context final HttpServletRequest httpReques
         return response;
     }
 
+    private Response addChildrenCountToResponse(final Response response, final User user, final PageMode pageMode, final String filter,
```

Comment:
I do not undestand why do you do this twice, if you want to return CategoryListDTO just implement a new Paginator

Reply:
I have added another column in the list so therefore I reused existing API of listing and just added children count in it. So I have created another DTO for it and returned all the old info including response headers

Reply:
I think would be clear to implement a CategoriesDTOPaginator

Reply:
I have implemented suggested changes :)